crypto/mem_sec.c

```diff
@@ -433,8 +438,19 @@ static int sh_init(size_t size, int minsize)
     if (mprotect(sh.map_result + aligned, pgsize, PROT_NONE) < 0)
         ret = 2;
 
+#if defined(OPENSSL_SYS_LINUX) && defined(MLOCK_ONFAULT) && defined(SYS_mlock2)
+    if (syscall(SYS_mlock2, sh.arena, sh.arena_size, MLOCK_ONFAULT) < 0) {
+        if (errno == ENOSYS) {
+            if (mlock(sh.arena, sh.arena_size) < 0)
+                ret = 2;
+        } else {
```

Comment:
> Based on my (limited) testing, the secure memory region when used with mlock() allocates it all and wires it

Looks like it, yes. I mean I can confirm this too. And it means that check for ENOSYS is [formally] sufficient. So it's still ready-to-go :-)